test/Log/LogServiceTest.php

```diff
@@ -302,6 +329,47 @@ public function testGetDefaultLog()
         $this->assertSame($logger, $logService->getDefaultLog());
     }
 
+    /**
+     * Test logger created by createLogger().
+     *
+     * @throws Exception
+     * @throws \ReflectionException
+     */
+    public function testCreateLogger()
+    {
+        $logService = $this->getLogService();
+
+        $logFilePath = $logService->getPath() . 'test.log';
+        $logFile = @fopen($logFilePath, 'a', false);
+        if ($logFile === false) {
+            $path = dirname($logFilePath);
+
+            if (! is_dir($path)) {
+                throw new \Exception('Directory for logging does not exist');
+            } else {
+                throw new \Exception('Failed to open logging file:' . $logFilePath);
+            }
+        }
+
+        $reflection = new \ReflectionClass($logService);
+
+        $createLogger = $reflection->getMethod('createLogger');
+        $createLogger->setAccessible(true);
+        $logger = $createLogger->invokeArgs($logService, ['%message%', 6, $logFile]);
```

Comment:
Please use the constant. Nobody knows what '6' means.

Reply:
We are still not testing if the format is prepared by `createLogger`. You could still comment out the call to `prepareFormat`.